CSLabs.Api/Controllers/TagController.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using CSLabs.Api.Models.ModuleModels;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace CSLabs.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class TagController : BaseController
+    {
+        public TagController(BaseControllerDependencies deps) : base(deps)
+        {
+        }
+        
+        [HttpGet]
+        public IActionResult Get()
+        {
+            return Ok(DatabaseContext.Tags.ToList());
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> Get(int id)
+        {
+            var tag = await this.DatabaseContext.Tags.FirstOrDefaultAsync(t => t.Id == id);
+            if (tag == null) 
+                return BadRequest();
+            return Ok(tag);
+        }
+
+        [HttpGet("/tag")]
```

Comment:
This should need a parameter capture syntax for it to work right? Has this been tested?
ex:
```
        [HttpGet("/tag/{tag}")]
```

Reply:
This has been tested and works, it is used for tag suggestions.